driver-core/src/main/com/mongodb/client/model/expressions/NumberExpression.java

```diff
@@ -19,46 +19,128 @@
 import java.util.function.Function;
 
 /**
- * Expresses a numeric value.
+ * A number value. {@link IntegerExpression Integers} are a subset of numbers,
+ * and so, for example, the integer 0 and the number 0 are the same value,
+ * and are equal.
  */
 public interface NumberExpression extends Expression {
```

Comment:
`IntegerExpression` explains the "integers are a subset of numbers" and appears to be the right place for such an explanation. Is it really useful to have that duplicated here?

Reply:
I thought it was not harmful to duplicate information (as a general convention in our docs), and helpful to note it here for those who might not read the Integer docs, since it is a notable deviation from how things work in Java.